tests/views/test_collection_exercise.py

```diff
@@ -198,503 +142,260 @@ class File:
+SEFT_SURVEY = {
+    "id": survey_id,
+    "longName": "Business Register and Employment Survey",
+    "shortName": "BRES",
+    "surveyRef": "221_201712",
+    "eqVersion": "",
+    "surveyMode": "SEFT",
+}
+CE_LINK = [sample_summary_id]
+SAMPLE_SUMMARY = {
+    "id": sample_summary_id,
+    "effectiveStartDateTime": "",
+    "effectiveEndDateTime": "",
+    "surveyRef": "",
+    "ingestDateTime": "2018-03-14T14:29:51.325Z",
+    "state": "ACTIVE",
+    "totalSampleUnits": 8,
+    "expectedCollectionInstruments": 1,
+    "areAllSampleUnitsLoaded": True,
+}
 
-        response = self.client.get(f"/surveys/{short_name}/{period}", follow_redirects=True)
+SAMPLE_SUMMARY_INIT = SAMPLE_SUMMARY.copy()
+SAMPLE_SUMMARY_INIT["state"] = "INIT"
 
-        self.assertEqual(response.status_code, 200)
-        self.assertIn("Monthly Survey of Building Materials Bricks".encode(), response.data)
-        self.assertIn("221_201712".encode(), response.data)
-        self.assertNotIn("Set as ready for live".encode(), response.data)
+CE_EVENTS = [
+    {"tag": "mps", "timestamp": "2018-10-11T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "go_live", "timestamp": "2018-10-11T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "return_by", "timestamp": "2018-10-30T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "reminder", "timestamp": "2018-10-12T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "exercise_end", "timestamp": "2018-10-13T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "nudge_email_0", "timestamp": "2018-10-14T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "nudge_email_1", "timestamp": "2018-10-14T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "nudge_email_2", "timestamp": "2018-10-15T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "nudge_email_3", "timestamp": "2018-10-16T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "ref_period_start", "timestamp": "2018-10-11T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+    {"tag": "ref_period_end", "timestamp": "2019-10-11T22:00:00.000+0000", "eventStatus": "PROCESSED"},
+]
 
-    @requests_mock.mock()
-    def test_collection_exercise_view_eq_ref_end_date(self, mock_request):
-        self.load_eq_survey(
-            mock_request,
-            self.eq_and_seft_survey,
-            self.collection_exercises,
-            collection_exercise_details["collection_exercise"],
-            self.collection_exercise_ref_end_date,
-            sample_summary_id,
-            self.sample_summary,
-            self.eq_collection_instrument,
-            self.eq_ci_selectors,
-        )
 
-        mock_request.get(url_get_by_survey_with_ref_end_date, json=collection_exercise_eq_ref_end_date)
+class TestCollectionExercise(ViewTestCase):
 
-        response = self.client.get(f"/surveys/{short_name}/{period}", follow_redirects=True)
+    @staticmethod
```

Comment:
The test class had a huge amount of boiler plate for each test, with patches, return_value, mocks etc. It made it very difficult to digest each test. Some just needed a couple of lines, but the boiler plate made everything cumbersome. This is another step towards pytest fixtures